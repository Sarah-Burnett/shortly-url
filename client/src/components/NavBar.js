import React, { Suspense, lazy } from "react";
import { Link } from "react-router-dom";
import Logo from "../images/logo.svg";
import styled from "styled-components";
import { breakpoint, alignPadding } from "./Styles";
import MediaQuery from "react-responsive";
import { Close, Burger } from "./Icons";
const NavLinks = lazy(() => import("./NavLinks"));

const Nav = styled.div`
	padding: 0 ${alignPadding};
	height: 10vh;
	display: flex;
	align-items: center;
	justify-content: space-between;
	background: #fff;
	.logo {
		height: 10vh;
		display: flex;
		align-items: center;
		img {
			max-height: 5vh;
		}
	}
`;
const NavBar = ({mobNav: {isMobNav, toggleMobNav}}) => {
	return (
		<Nav>
			<Link className="logo" to="/">
				<img src={Logo} alt="Shortly" />
			</Link>
			<Suspense fallback={<div />}>
				<MediaQuery minDeviceWidth={breakpoint}>
					<NavLinks />
				</MediaQuery>
				<MediaQuery maxDeviceWidth={breakpoint}>
					{isMobNav && <NavLinks toggleMobNav={toggleMobNav} />}
				</MediaQuery>
			</Suspense>
			<MediaQuery maxDeviceWidth={breakpoint}>
				<div onClick={toggleMobNav}>{isMobNav ? <Close /> : <Burger />}</div>
			</MediaQuery>
		</Nav>
	);
};

export default NavBar;
