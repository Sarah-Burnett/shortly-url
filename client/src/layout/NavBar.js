import React, { useState, useContext } from 'react';
import { UserContext } from '../context/UserContext';
import { Link } from 'react-router-dom';
import Logo from '../images/logo.svg';
import styled from 'styled-components';
import { breakpoint, Button, PrimaryButton, white, grey } from '../Styles';
import MediaQuery from 'react-responsive';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';

const Nav = styled.div`
    width: 100vw;
    height: 10vh;
    padding: 0 8vw;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: #FFF;
    @media (min-width: ${breakpoint}) {
        padding: 0vh 8vw;
    }
`
const NavBtn = styled(Button)`
    margin: 1vw;
`

const MobNavContainer = styled.div`
    height: 60vh;
    width: 100vw;
    display: flex;
    align-items: flex-start;
    justify-content: center;
    background: ${white};
`
const MobNav = styled.div`
    height: 100%;
    width: 80%;
    background: hsl(257, 27%, 26%);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
    text-align: center;
    text-weight: bold;
    border-radius: 15px;
    button {
        color: ${white};
        font-size: larger;
    }
    hr{
        width: 75%;
        border-bottom: 1px solid ${grey};
        }
    }
    .signup {
        width: 75%;
        border-radius: 30px;
        padding: 2vh 2vw;
    }
`

export const NavBar = () => {
    const [ user, setUser ] = useContext(UserContext);
    const { isAuth } = user;
    const [isMobNav, setIsMobNav] = useState(false);
    const toggleMobNav = () => isMobNav ? setIsMobNav(false) : setIsMobNav(true);
    return (
        <nav>
            <Nav>
                <Link to="/"><img src={Logo} alt="Shortly"/></Link>
                <MediaQuery minDeviceWidth={breakpoint}>
                    <div style={{flexBasis: "50vw"}}>
                        <NavBtn>Features</NavBtn>
                        <NavBtn>Pricing</NavBtn>
                        <NavBtn>Resources</NavBtn>
                    </div>
                    <div>
                    { !isAuth && <NavBtn><Link to="/login">Login</Link></NavBtn> }
                    { !isAuth && <PrimaryButton><Link to="/signup">Sign Up</Link></PrimaryButton>}
                    { isAuth && <PrimaryButton><Link to="/logout">Logout</Link></PrimaryButton>}
                    </div>
                </MediaQuery>
                <MediaQuery maxDeviceWidth={breakpoint}>
                    <FontAwesomeIcon icon={faBars} style={{color: "hsl(0, 0%, 75%)"}} onClick={toggleMobNav}/>
                </MediaQuery>
            </Nav>
            {isMobNav && <MobNavContainer>
                <MobNav>
                    <NavBtn>Features</NavBtn>
                    <NavBtn>Pricing</NavBtn>
                    <NavBtn className="resources">Resources</NavBtn>
                    <hr/>
                    { !isAuth && <NavBtn><Link to="/login">Login</Link></NavBtn> }
                    { !isAuth && <PrimaryButton><Link to="/signup">Sign Up</Link></PrimaryButton>}
                    { isAuth && <PrimaryButton><Link to="/logout">Logout</Link></PrimaryButton>}
                </MobNav>
            </MobNavContainer>}
        </nav>
    );
}
