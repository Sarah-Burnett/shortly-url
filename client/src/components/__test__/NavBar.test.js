import React from "react";
import ReactDOM from "react-dom";
import NavBar from "../NavBar";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

const NavBarWrapper = () => (
	<Router>
		<NavBar />
	</Router>
);

it("renders NavLinks without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<NavBarWrapper />, div);
});

it("nav bar matches snapshot", () => {
	const tree = renderer.create(<NavBarWrapper />).toJSON();
	expect(tree).toMatchSnapshot();
});

//TODO: add check for media query ??
