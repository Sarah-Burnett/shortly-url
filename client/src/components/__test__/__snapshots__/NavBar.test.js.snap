// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar matches snapshot 1`] = `
<div
  className="sc-fzoLsD bJNbyp"
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Shortly"
      src="logo.svg"
    />
  </a>
</div>
`;
